gammapy/spectrum/spectrum_analysis.py

```diff
@@ -23,84 +28,126 @@ def main(args=None):
-                nruns = nruns + 1
+                temp = SpectrumObservation(val, self.store, on_region,
+                                           bkg_method, ebounds, exclusion)
+            except IndexError:
+                log.warn(
+                    'Observation {} not in store {}'.format(val, datastore))
+                nobs = nobs + 1
                 continue
-            self.observations.append(val)
-            if i == nruns:
+            self._observations.append(temp)
+            if i == nobs - 1:
                 break
 
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def info(self):
-        log.info(self.store.info())
-        log.info('ON region\nCenter: {0}\nRadius: {1}'.format(
-            self.target, self.radius))
-        log.info('OFF region\nType: {0}\nInner Radius: {1}\nOuter Radius: {2}'.format(
-            'Ring', self.irad, self.orad))
-
-    def run(self):
-        """Run analysis as specified in the config"""
-        if self.config['general']['create_ogip']:
-            self.make_ogip()
-        if self.config['general']['run_fit']:
-            fit = self.run_fit()
-            return fit
-
-    def make_ogip(self):
-        """Create OGIP files"""
+        if len(self.observations) == 0:
+            raise ValueError("No valid observations found")
+
+    @property
+    def observations(self):
+        """List of all observations belonging to the analysis
+        """
+        return self._observations
+
+    @property
+    def reflected_regions(self, **kwargs):
+        """List of dicts containing information about the reflected regions
+        for each observation
+        """
+        retval = list([])
+        for obs in self.observations:
+            reflected = obs.make_reflected_regions(**kwargs)
+            val = dict(obs=obs.obs, pointing=obs.pointing, region=reflected)
+            retval.append(val)
+        return retval
+
+    def info(selfs):
+        """Print some information
+        """
+        pass
+
+    def write_ogip_data(self, dir=None):
+        """Create OGIP files
+
+        Parameters
+        ----------
+        dir : str (optional)
+            write directory
+        """
+        if dir is None:
+            dir = 'ogip_data'
+
         for obs in self.observations:
-            obs.make_ogip()
+            obs.write_all_ogip_data(dir)
             log.info('Creating OGIP data for run{}'.format(obs.obs))
 
-    def run_fit(self):
-        """Run the gammapy.hspec fit"""
+    def run_hspec_fit(self, model, thres_low, thres_high):
+        """Run the gammapy.hspec fit
+
+        Parameters
+        ----------
+        model : str
+            Sherpa model
+        thres_high : `~gammapy.spectrum.Energy`
```

Comment:
Combine `thres_high` and `thres_low` into a `energy_range` tuple?
If you don't like that, put low before high here in the docstring, to be consistent with the order of arguments in the function.